ldms/src/ldmsd/ldmsd.h

```diff
@@ -1028,12 +1041,12 @@ typedef struct ldmsd_sampler_set {
 
 struct ldmsd_cfgobj_sampler {
 	struct ldmsd_cfgobj cfg;
+	struct ldmsd_plugin_generic *plugin;
 	struct ldmsd_sampler *api;
 
 	/* Set to 1 if the plugin has been configured */
 	int configured;
 
-	char *libpath;
 	/* Private context pointer, managed by plugin */
 	void *context;
```

Comment:
Should the context pointer be moved to the ldmsd_plugin_generic structure too?

Reply:
Actually, thinking about this again, I wonder if it wouldn't be better to leave this where it is.

Are we always going to keep the code organized the way it is, with plugins redundantly opened for each instance? We really only need to dlopen once per plugin, and instances can share the same information. It would be easier to stop redundantly opening plugins in the future if we keep per-instance stuff in the per-instance cfgobj (e.i. cfgobj_sampler, cfgob_store), and let the generic plugin struct remain specific to the generic plugin information.